src/main/java/uk/gov/hmcts/reform/wataskmonitor/services/jobs/failure/initiation/TaskInitiationFailuresJobService.java

```diff
@@ -0,0 +1,129 @@
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.wataskmonitor.clients.CamundaClient;
+import uk.gov.hmcts.reform.wataskmonitor.config.job.InitiationJobConfig;
+import uk.gov.hmcts.reform.wataskmonitor.domain.camunda.CamundaTask;
+import uk.gov.hmcts.reform.wataskmonitor.domain.camunda.CamundaVariable;
+import uk.gov.hmcts.reform.wataskmonitor.domain.jobs.GenericJobOutcome;
+import uk.gov.hmcts.reform.wataskmonitor.domain.jobs.GenericJobReport;
+import uk.gov.hmcts.reform.wataskmonitor.utils.LoggingUtility;
+import uk.gov.hmcts.reform.wataskmonitor.utils.ResourceUtility;
+
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import static java.util.Collections.emptyList;
+import static uk.gov.hmcts.reform.wataskmonitor.domain.taskmonitor.JobName.TASK_INITIATION_FAILURES;
+import static uk.gov.hmcts.reform.wataskmonitor.services.ResourceEnum.CAMUNDA_TASKS_UNINITIATED;
+
+@Component
+@Slf4j
+public class TaskInitiationFailuresJobService {
+
+    public static final String CAMUNDA_DATE_REQUEST_PATTERN = "yyyy-MM-dd'T'HH:mm:ss.SSSZ";
+    private final DateTimeFormatter formatter = DateTimeFormatter.ofPattern(CAMUNDA_DATE_REQUEST_PATTERN);
+
+    private final CamundaClient camundaClient;
+    private final InitiationJobConfig initiationJobConfig;
+
+    @Autowired
+    public TaskInitiationFailuresJobService(CamundaClient camundaClient,
+                                            InitiationJobConfig initiationJobConfig) {
+        this.camundaClient = camundaClient;
+        this.initiationJobConfig = initiationJobConfig;
+    }
+
+    public GenericJobReport getUnInitiatedTasks(String serviceToken) {
+        log.info("{} initiationJobConfig: {}", TASK_INITIATION_FAILURES.name(), initiationJobConfig.toString());
+        List<CamundaTask> camundaTasks = camundaClient.getTasks(
+            serviceToken,
+            "0",
+            initiationJobConfig.getCamundaMaxResults(),
+            buildSearchQuery()
+        );
+
+        log.info("{} {} task(s) retrieved successfully.",
+            TASK_INITIATION_FAILURES.name(), camundaTasks.size());
+
+        if (camundaTasks.isEmpty()) {
+            log.info("{} There was no task", TASK_INITIATION_FAILURES.name());
+            return new GenericJobReport(0, emptyList());
+        } else {
+            log.warn("{} There are some uninitiated tasks", TASK_INITIATION_FAILURES.name());
+            List<GenericJobOutcome> outcomesList = prepareUnInitiatedTasksAndReturnOutcome(camundaTasks, serviceToken);
+            return new GenericJobReport(camundaTasks.size(), outcomesList);
+        }
+
+    }
+
+    private List<GenericJobOutcome> prepareUnInitiatedTasksAndReturnOutcome(List<CamundaTask> camundaTasks,
```

Comment:
This seems like it should not belong here.
I would expect that either something very similar to this is being done elsewhere that you can re-use or this should be in its own class so that it can be re-used another time for another Job.

Reply:
raise a ticket for duplicated methods.